module/zfs/arc.c

```diff
@@ -9337,26 +9337,27 @@ l2arc_apply_transforms(spa_t *spa, arc_buf_hdr_t *hdr, uint64_t asize,
 	}
 
 	if (compress != ZIO_COMPRESS_OFF && !HDR_COMPRESSION_ENABLED(hdr)) {
-		cabd = abd_alloc_for_io(asize, ismd);
-		tmp = abd_borrow_buf(cabd, asize);
+		cabd = abd_alloc_for_io(size, ismd);
+		tmp = abd_borrow_buf(cabd, size);
 
 		psize = zio_compress_data(compress, to_write, tmp, size,
 		    hdr->b_complevel);
 
-		if (psize >= size) {
-			abd_return_buf(cabd, tmp, asize);
```

Comment:
If `zio_compress_data()` fails to return `psize < size`, then we should abort compression and return an appropriate buffer to be cached in L2ARC. Given that above panic implies that `size > asize`, it would make sense here to make the proposed change.

In the general case of no encryption, no compression, the buffer will have `asize >= size`. I do not think this is going to cause any problems.

Reply:
I cannot tell what you're trying to communicate in this comment.

Reply:
I am just putting my thoughts down, I think the proposed change from `if (psize >= size)` to `if (psize >= asize)` makes sense, and cannot see any problems with it.

Reply:
Okay, great; I asked because that's how I read it, but I usually don't see review comments that don't require changes, so I assumed I was failing to read it correctly.